core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -489,24 +491,63 @@ void toSerializable(DistributedStorage storage)
     // Get set is re-validated to check if there is no anti-dependency
     for (Map.Entry<Get, Optional<TransactionResult>> entry : getSet.entrySet()) {
       Get get = entry.getKey();
-      Key key = new Key(get);
-      if (writeSet.containsKey(key) || deleteSet.containsKey(key)) {
-        continue;
-      }
 
-      tasks.add(
-          () -> {
-            Optional<TransactionResult> originalResult = getSet.get(get);
-            // Only get the tx_id column because we use only them to compare
-            get.clearProjections();
-            get.withProjection(Attribute.ID);
-
-            // Check if a read record is not changed
-            Optional<TransactionResult> latestResult = storage.get(get).map(TransactionResult::new);
-            if (isChanged(latestResult, originalResult)) {
-              throwExceptionDueToAntiDependency();
-            }
-          });
+      if (ScalarDbUtils.isSecondaryIndexSpecified(get, getTableMetadata(get))) {
+        // For Get with index
+
+        tasks.add(
+            () -> {
```

Comment:
Can we create a method for this part for consistency?
Something like this.

```
tasks.add(() -> validateGetWithIndexResults(...));
```

Also, the other part can be done in the same way?

```
tasks.add(() -> validateGetResults(...)); // `validateWithoutGetResults` is an option.
```

Reply:
Fixed in e00777a377ab7f6e2f13b30b3a39b9f4e5b25547. Thanks!